core/modules/config.go

```diff
@@ -30,6 +30,9 @@ type ModuleConfig struct {
 
 	// The version of the engine this module was last updated with.
 	EngineVersion string `json:"engineVersion,omitempty"`
+
+	// Whether to automatically generate a .gitignore file for this module.
+	AutomaticGitignore *bool `json:"automaticGitignore,omitempty"`
```

Comment:
Don't want to drown this in *too* much bikeshedding! But I wonder if this makes sense under a sub-object, `codegen.automaticGitignore`? Couple of reasons:

- I think it would be nice to keep the top-level clean if we can
- I can easily imagine adding more codegen-related settings one-day
- For some reason, it feels *different* from the other options in here - which are more about *loading* the module, instead of *developing* it.

Reply:
Sure, it's a little awkward as the only setting in that object, but I suppose that's always going to be the case for the first thing we put in there. Updated to put it in a "codegen" object